lib/Sema/CSGen.cpp

```diff
@@ -2293,8 +2293,10 @@ namespace {
         return setType(
             externalPatternType
                 ? externalPatternType
-                : CS.createTypeVariable(CS.getConstraintLocator(locator),
-                                        TVO_CanBindToNoEscape));
+                : CS.createTypeVariable(
+                      CS.getConstraintLocator(
+                          pattern, ConstraintLocator::AnyPatternDecl),
+                      TVO_CanBindToNoEscape | TVO_CanBindToHole));
```

Comment:
Since you’re at it, I think this would read easier as

```cpp
        Type type;
        if (externalPatternType) {
          type = externalPatternType;
        } else {
          type = CS.createTypeVariable(CS.getConstraintLocator(
                                           pattern, ConstraintLocator::AnyPatternDecl),
                                       TVO_CanBindToNoEscape | TVO_CanBindToHole);
        }
        return setType(type);
```

Reply:
Sure, I can do that!

Reply:
Done!